app/controllers/theme_creator/theme_creator_controller.rb

```diff
@@ -241,6 +242,16 @@ def ban_for_remote_theme!
     # no-op - we want to allow this stuff for theme-creator
   end
 
+  def objects_setting_metadata
+    theme = Theme.find_by(id: params[:id])
+    raise Discourse::InvalidParameters.new(:id) unless theme
+
+    theme_setting = theme.settings[params[:setting_name].to_sym]
+    raise Discourse::InvalidParameters.new(:setting_name) unless theme_setting
+
+    render_serialized(theme_setting, ThemeObjectsSettingMetadataSerializer, root: false)
+  end
+
```

Comment:
This looks identical to the method in `Admin::ThemesController`, which is the parent class of `ThemeCreator::ThemeCreatorController`. So I think we can drop this copy/paste?

```suggestion
```

Reply:
ah yes sounds good